### PR TITLE
refactor(various): migrate to foundation & BEM scss

### PR DESCRIPTION
**Pull request by ncoden**
Update form for the web-client-ng BEM refactor.
See https://github.com/CaliOpen/caliopen.web-client-ng/pull/63

### CODE REVIEW

**Comment by ncoden** [comment]
@iamdey

File: app/views/login.html [hunk 1]
```diff
@@ -1,49 +1,49 @@
...
+              <span class="sr-only">Error:</span>
+              {{error}}
+            </div>
+            {% endif %}
+
+            <p class="s-login__form-row">
+              <label for="username" class="s-login__label">
+                Username
+              </label>
+              <input type="text" name="username" placeholder="Username" class="s-login__form-input m-text-field m-text-field--light" />
```
Comment (iamdey) [inline app/views/login.html]: it would be great if you add the missing id for the input

**Comment by iamdey** [comment]
can you rebase on master please?

File: app/views/login.html (hunk 1 above)
Comment (ncoden) [inline app/views/login.html]: The missing id. I don't understand

Comment (iamdey) [inline app/views/login.html]: label using for attribute : http://devdocs.io/html/element/label

File: app/views/login.html [hunk 2]
```diff
@@ -1,49 +1,49 @@
...
-            </label>
-            <input type="password" name="password" class="form-control cor-login__form-input" />
-          </p>
-
-          <input type="submit" class="btn btn-primary btn-lg btn-block cor-login__form-submit" value="Go" />
+          <div class="m-panel__content">
+            {% if error %}
+            <div class="callout alert" role="alert">
+              <span class="fa fa-exclamation-circle" aria-hidden="true"></span>
+              <span class="sr-only">Error:</span>
```
Comment (iamdey) [inline app/views/login.html]: `sr-only` doesn't exist anymore. it's `show-for-sr` in foundation: http://foundation.zurb.com/sites/docs/visibility.html#accessibility

File: app/views/login.html [hunk 3]
```diff
@@ -1,49 +1,49 @@
...
-            <label for="password"  class="cor-login__label">
-              Password
-            </label>
-            <input type="password" name="password" class="form-control cor-login__form-input" />
-          </p>
-
-          <input type="submit" class="btn btn-primary btn-lg btn-block cor-login__form-submit" value="Go" />
+          <div class="m-panel__content">
+            {% if error %}
+            <div class="callout alert" role="alert">
```
Comment (iamdey) [inline app/views/login.html]: `callout` is imported from foundation but not `alert` :(

Comment (iamdey) [inline app/views/login.html]: In the BEM way, you confirm we need a `m-callout` and its modifiers `m-callout--alert` etc. ?

File: app/views/login.html (hunk 2 above)
Comment (ncoden) [inline app/views/login.html]: I know. Just a mistake :)

File: app/views/login.html (hunk 3 above)
Comment (ncoden) [inline app/views/login.html]: True. But we have these component that work, let's use them as they are and not complicating things.
For margins, we can have:
- `s-login__m-callout` if it is a very specific case (but it isn't)
- `m-form__item` which ensure than each field or callout have enough space around it.
- `--margin` or `--listed` which can set `margin-right` or `margin-bottom` only. I think is the better solution, but it's not 	convenient.

All component we define shouldn't have default margins, but should have it when we put them next to something. With inheritance possibilities, we could define a `o-component` (`o-` for object, abstract) with all these basic features, from all component inherit.

What do you think ?

**Comment by ncoden** [comment]
@iamdey Fixed

File: app/views/login.html (hunk 3 above)
Comment (iamdey) [inline app/views/login.html]: I'm not sure I understand the `o-` thing. 

Anyway, I'm ok with your solutions. you choose.

Comment (ncoden) [inline app/views/login.html]: `o-`: abstract object. A set of usefull coherent features (still object-oriented), but not done to be directly used in HTML as a component (no theme, no padding/margin or size, no semantic).
Inspired from: http://csswizardry.com/2015/03/more-transparent-ui-code-with-namespaces/
*csswizardry* have a lot of very good articles, take a look !

So, we can define this abstract object which contain "the behavior and basic features we want on all our components", and make our component inherit from it. *It require the OO mixins I pushed on the [bem-utils branch](https://github.com/ncoden/caliopen.web-client-ng/tree/refactor/bem-utils)*

I prefer by far this last *giga-object-oriented* solution, but it need lot of more work :)
I'm currently working on it in an other repo.

**Comment by iamdey** [comment]
@ncoden last thing, I promise: can you add some space before the error callout ? 

capture du 2016-04-20 11 50 27

**Comment by ncoden** [comment]
@iamdey done :)

*You promised*

**Pull request merged**